[Yu Jiaxiang] iP

litone01 opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

litone01 committed:
add level 1

Finished requirement for level 1.

Duke now starts by greeting the user, then echos commands entered by the user,
and exits when the user types `bye`.

litone01 committed:
add level 2

Add the ability to store whatever text entered by the user
and display them back to the user when requested.

litone01 committed:
Revert "add level 2"

This reverts commit 7f8a81f37d3392af14b9a014cd3c1fb8c6a68025.

litone01 committed:
add level 2 features

Add the ability to store whatever text entered by the user
and display them back to the user when requested.

litone01 committed:
Merge pull request #1 from litone01/heads/Level-2

add level 2 features

litone01 committed:
add level 3 features

1. Level-3 feature: add the ability to mark tasks as done.
2. Handle two errors when implementing the function of add/mark task as done.

litone01 committed:
Fix minor format issues for level-3 feature implementation

litone01 committed:
Add level-4 features

Add support for tracking three types of tasks:
1. ToDos: tasks without any date/time attached to it
2. Deadlines: tasks that need to be done before a specific date/time
3. Events: tasks that start at a specific time and ends at a specific time

Coming up:
1. A lot of error and exception handling is needed :(

litone01 committed:
Merge pull request #2 from litone01/level-4

Add level-4 features

litone01 committed:
Add Level-5

Handle errors when user enters incorrect inputs.  For example (not exclusive):
1. When users create a todo without providing a description.
2. When users enters a command word which is not supported by the current iteration of duke

litone01 committed:
Add Level-6

Add support for deleting tasks from the list.

litone01 committed:
(Complete for this iteration) Add A-TextUiTesting

* (Incomplete) Add A-TextUiTesting

Use the input/output redirection technique to semi-automate the testing of Duke.

* (Complete for this iteration) Add A-TextUiTesting

Extend input and expected to test all functionalities and exceptions designed.

litone01 committed:
Fix bug for incorrect display of the status icon

litone01 committed:
Add level-7

Save the tasks in the hard disk automatically whenever the task list changes.
Load the data from the hard disk when Duke starts up.
Filename and file location is hard coded to src/main/java/duke/data/duke.txt

litone01 committed:
Minor Path Update for save

litone01 committed:
Update textui-test for level-7

litone01 committed:
Add level-8

Achieve minimal goal.
1. Store deadline dates as a java.time.LocalDate.
2. Accept dates in "yyyy-mm-dd" format and print in "MMM dd yyyy"
3. Any invalid input or other format will be caught as an exception

litone01 committed:
Merge pull request #4 from litone01/branch-level-7

Merged Branch level 7

litone01 committed:
Merge branch 'branch-level-8'

Resolved Conflicts in:
#	src/main/java/duke/Duke.java

litone01 committed:
Fix bug when merging branch-level-7 and branch-level-8 to master

Saving and accepting date as input for deadline are now working.

litone01 committed:
Update README.md

litone01 committed:
Update README.md

litone01 committed:
add A-MoreOOP

Refactor the code to extract out closely related code as classes.
Ui: deals with interactions with the user
Storage: deals with loading tasks from the file and saving tasks in the file
Parser: deals with making sense of the user command
TaskList: contains the task list e.g., it has operations to add/delete tasks in the list

To be done:
Seperate tasklists from storage
Avoid using static storage for commands

Disclaimer: this commit is heavily referenced to the folder structure and coding style of se-edu/addressbook-level2

litone01 committed:
Add disclaimer for folder structure and code style reference

litone01 committed:
Add A-JUnit

Add JUnit test for Task and Todo class, very simple examples as a start to learn about testing.

litone01 committed:
Add Javadocs for all public classes and methods

Todo: add javadocs for the classes for testing.

litone01 committed:
Add Javadoc for UI class

litone01 committed:
Comply with coding standard

Tweak the code to comply with a given coding standard.

litone01 committed:
Add level-9, find function

Allows users to find a task by searching for a keyword.

litone01 committed:
Merge pull request #7 from litone01/branch-A-JavaDoc

Added JavaDoc for all public class and methods except testing

litone01 committed:
Merge pull request #8 from litone01/branch-A-CodingStandard

Comply with coding standard

litone01 committed:
Merge pull request #9 from litone01/branch-level-9

Add level-9, find function

litone01 committed:
Merge pull request #10 from litone01/add-gradle-support

Merged Add gradle support

litone01 committed:
Add Gradle Support and fixed checkstyle errors found

Current version of the project is compatible with gradle for more automation.

Gradle support can be added and code style can be rectified.

Let's add the support and fix the checkstyle issues

litone01 committed:
Merge pull request #11 from litone01/A-Gradle

Add Gradle Support and fixed checkstyle errors found

litone01 committed:
Add level-10 GUI features

The current version only supports CLI.

GUI supports can be added.

Let's add GUI support!

Something that may be improved:
1. Fix issues of dispalying incomplete list when the list is very long
2. Better organisation of code with load ui and storage for gui

litone01 committed:
Fix checkstyle issue

guanyz commented on src/main/java/duke/ui/UI.java:
```diff
@@ -0,0 +1,89 @@
+/**
+ * Interacts with user by receiving input and printing results from user commands.
+ */
+public class UI {
```
Should this be in PascalCase (Ui)?

litone01 replied:
Thanks for the point! I think I will check with the coding standard again to see if it should be Ui or UI. Personally find it more natural to use UI hahaha.

guanyz commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,69 @@
+        new Duke().run(args);
+    }
+
+    private void run(String[] args) {
+        start();
+        runUntilExit();
+        exit();
+    }
```
I like how clean this method is!

litone01 replied:
Thanks!

guanyz commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,152 @@
+     * @return the information of the tasks present
+     */
+    public static String[] getAllTaskListInfo() {
+        String[] res;
```
Perhaps this variable could be given a clearer name? It took me a bit to realize what it was meant to store.

litone01 replied:
Yep sure! I think it is a valid point!

guanyz reviewed (commented):
Nice implementation of Duke! The code looks nice and clean overall.

litone01 committed:
Merge pull request #12 from litone01/branch-level-10

Add level 10 GUI

The current version only supports CLI.

GUI supports can be added.

Let's add GUI support!

Something that may be improved:
1. Fix issues of dispalying incomplete list when the list is very long
2. Better organisation of code with load ui and storage for gui

litone01 committed:
Fix bug of shortened display of list

litone01 committed:
Build GUI jar for release

litone01 committed:
Update file structure and filename

Deadline, event, todo, tasklist are not under package task. Exception is also in the same package with non-relevant classes.

Regroup and rename relevant classes together.

litone01 committed:
Fix checkstyle issues

litone01 committed:
Add assertions for tasks

Assertions are added to check when tasks are created and their information stored,
the relevant information such as description, completion date(if applicable) are present.

litone01 committed:
Refactor for a better code quality

Codes can be adjusted to fit the code quality as required.

Try to follow various code quality requirements,
including SLAP, no arrow head, better readability, better name for variables and functions.

litone01 committed:
Merge pull request #13 from litone01/branch-A-Assertions

Add assertions for tasks

litone01 committed:
Merge pull request #14 from litone01/branch-A-CodeQuality

Refactor for a better code quality

Codes can be adjusted to fit the code quality as required.

Try to follow various code quality requirements,
including SLAP, no arrow head, better readability, better name for variables and functions.

litone01 committed:
Update .gitignore file to ignore new data folder

litone01 committed:
Merge pull request #15 from litone01/branch-A-CodeQuality

Update .gitignore file to ignore new data folder

litone01 committed:
Update name of UI to Ui to follow coding standard

litone01 committed:
Merge pull request #16 from litone01/branch-A-CodeQuality

Update name of UI to Ui to follow coding standard

litone01 committed:
Add CI for push and pull request

litone01 committed:
Merge branch 'master' of https://github.com/litone01/ip

litone01 committed:
Removed CI

litone01 committed:
Create gradle.yml

litone01 committed:
Fix checkstyle issues

litone01 committed:
Possible fix for building error

litone01 committed:
Fix building error for I/O testing

litone01 committed:
Merge pull request #17 from litone01/branch-CI

Add CI for push and pull request

litone01 committed:
Add update for deadline

Deadline should be able to be updated given a new by date

Now deadlines can be updated with a new by date (without being deleted and recreated)
syntax is as follows: "update 1 2020-01-01"
1 is the index of the deadline to be updated as according to the list
2020-01-01 is the new by date in yyyy-mm-dd format

litone01 committed:
Update expected.txt for i/o testing

litone01 committed:
Merge pull request #18 from litone01/branch-C-Update

Add update for deadline

Deadline should be able to be updated given a new by date

Now deadlines can be updated with a new by date (without being deleted and recreated)
syntax is as follows: "update 1 2020-01-01"
1 is the index of the deadline to be updated as according to the list
2020-01-01 is the new by date in yyyy-mm-dd format

litone01 committed:
Update README.md

litone01 committed:
Improved the GUI

Background and user images can be changed.
Dialogs can be further away from the user or duke image.

Add background image and change the duke and user profile image.
Paddings are added to adjust the dialogs.

litone01 committed:
Update stylesheet name

litone01 committed:
Merge pull request #20 from litone01/A-BetterGui

Improved the GUI

litone01 committed:
Minor updates on javadoc

Makes the description of the updateCommand more specific.

litone01 committed:
Add User Guide

User guide can be added to the project.

Adds user guide.

litone01 committed:
Fix checkstyle issues

Fix checkstyle issues detected by not having '+' on a new line

litone01 committed:
Update expected.txt after updating the result message of update

litone01 committed:
Set theme jekyll-theme-cayman

litone01 committed:
Update image quality and add explanation for exit

litone01 committed:
Merge pull request #21 from litone01/A-UserGuide

Add user guide

litone01 committed:
Set theme jekyll-theme-slate

litone01 committed:
Set theme jekyll-theme-cayman

litone01 committed:
Modify codes for better code quality

Codes can be improved for better quality.

Modifications are done to various files.

litone01 committed:
Merge pull request #22 from litone01/branch-finalize

Modify codes for better code quality
